src/XliffTasks/build/XliffTasks.targets

```diff
@@ -56,18 +56,23 @@
   </Target>
 
   <Target Name="UpdateXlfOnBuild"
-          Condition="'$(EnableXlfLocalization)' == 'true' and '$(UpdateXlfOnBuild)' == 'true' and '$(_IsInnerXlfBuild)' != 'true'"
+          Condition="'$(EnableXlfLocalization)' == 'true'
+                     and '$(UpdateXlfOnBuild)' == 'true'
+                     and '$(_IsInnerXlfBuild)' != 'true'
+                     and '$(BuildingForLiveUnitTesting)' != 'true'
+                     and '$(DesignTimeBuild)' != 'true'
+                     and '$(BuildingProject)' != 'false'"
```

Comment:
Suspect the last two don't run BeforeBuild (do they?). It's fine to be defensive though.

Reply:
Do we also need to turn off EnsurexlfIsUpToDate? I could imagine that failing in these builds once we skip the Update.

Reply:
@nguerrera You mean design-time builds don't run the `BeforeBuild` target? Probably not, but as you say it's fine to be defensive. On the other hand, if design-time builds don't end up running these targets anyway, then there's no need for this change.

Reply:
No, we still need this. Some design-time builds run the `Build` target, which runs `BeforeBuild`, which runs `UpdateXlfOnBuild`.

We need the same check on `EnsurexlfIsUpToDate`, or perhaps `TranslateSouceFromXlf` would be better.

Also, should we be excluding Live Unit Testing? I can well imagine that unit tests may depend on the satellite assemblies...

Reply:
I can imagine that unit tests depend on satellites, but updating the XLF will just update English text which should have no functional difference unless your unit test is asserting that a resource was found in the satellite and not the fallback, but that seems very, very rare. Maybe we could have an option -- UpdateXlfOnLiveUnitTestingBuild -- defaulted to false?

Reply:
No, you're right. Let's just exclude LUT until someone complains. :-)